queries/slippage/evaluate_slippage_from_internal_token_transfers_per_tx.sql

```diff
@@ -0,0 +1,84 @@
+,
+final_token_balance_sheet as (
+    select
+        solver_address,
+        solver_name,
+        sum(amount_from) token_imbalance_wei,
+        symbol,
+        token_from,
+        tx_hash
+    from
+        incoming_and_outcoming_with_buffer_trades
+    group by
+        symbol,
+        token_from,
+        solver_address,
+        solver_name,
+        tx_hash
+),
+end_prices as (
+    select
+        median_price as price,
+        p_complete.contract_address,
+        decimals
+    from
+        prices.prices_from_dex_data p_complete
+    where
+        p_complete.hour = '{{EndTime}}'
+),
+results as (
+    select
+        solver_address,
+        solver_name,
+        sum(token_imbalance_wei * price / 10 ^ p.decimals) as usd_value,
+        tx_hash
+    from
+        final_token_balance_sheet
+        inner join end_prices p on token_from = p.contract_address
+    group by
+        solver_address,
+        solver_name,
+        tx_hash
+    having
+        sum(token_imbalance_wei) != 0
+)(
+    select
+        *,
+        (
+            usd_value / (
+                select
+                    price
+                from
+                    prices."layer1_usd_eth"
+                where
+                    minute = '{{EndTime}}'
+            ) * 10 ^ 18
+        ) as eth_slippage_wei
+    from
+        results
+    order by
+        usd_value ASC
+    limit
+        5
+)
+UNION
+all (
+    select
+        *,
+        (
+            usd_value / (
+                select
+                    price
+                from
+                    prices."layer1_usd_eth"
+                where
+                    minute = '{{EndTime}}'
+            ) * 10 ^ 18
+        ) as eth_slippage_wei
+    from
+        results
+    order by
+        usd_value DESC
+    limit
+        5
```

Comment:
Ok I see what you are doing here and would suggest the following. 

Don't duplicate this query just to take the top and bottom 5 results. Instead fetch everything with the first query and then take a slice of the sorted records in python.


```py
results = sorted([Class(record) for record in dune.fetch()], key= lambda t: t.usd_value))
# Top and bottom 5 results.
slice = results[:5] + results[-5:]
```

Reply:
We need to have a different query. The one is grouping by tx_hash, and the other one is not.

Reply:
I would really like to get the hard coded limits of 5 records out of this query. Above I suggested how you can get the top and bottom 5 without having to limit the query. Also, this query seems to be only for the purpose of testing it could be moved to the e2e directory.